.github/workflows/cypress_workflow.yml

```diff
@@ -6,16 +6,43 @@ on:
     branches: [ '**' ]
     paths-ignore:
       - '**/*.md'
+  workflow_dispatch:
+    inputs:
+      test_repo:
+        description: 'Test repo'
+        default: 'opensearch-project/opensearch-dashboards-functional-test'
+        required: true
+        type: string
+      test_branch:
+        description: 'Test branch (default: source branch)'
+        required: false
+        type: string
+      pr_number:
+        description: 'PR Number'
+        required: false
+        type: number
+      specs:
+        description: 'Additional tests to run'
+        required: false
+        type: string
 
 env:
+  SOURCE_REPO: ${{ github.repository }}
+  SOURCE_BRANCH: ${{ github.base_ref }}
+  TEST_REPO: ${{ inputs.test_repo != '' && inputs.test_repo || 'opensearch-project/opensearch-dashboards-functional-test' }}
+  TEST_BRANCH: "${{ inputs.test_branch != '' && inputs.test_branch || github.base_ref }}"
```

Comment:
If a PR is raised against a feature branch, `github.base_ref` would point to that branch; what if that branch doesn't exist on the FT repo?

Reply:
yeah. that's happened a couple of times in the pass.

i end up just creating the branch in the FTRepo. Part of the release improvements and artifacts we want to have fallback solutions.